Index: quagga-1.2.1/lib/if.c
===================================================================
--- quagga-1.2.1.orig/lib/if.c
+++ quagga-1.2.1/lib/if.c
@@ -908,7 +908,16 @@ connected_add_by_prefix (struct interfac
                          struct prefix *destination)
 {
   struct connected *ifc;
+  struct listnode *cnode;
+  struct connected *c;
+  int ret = 0;

+  for (ALL_LIST_ELEMENTS_RO (ifp->connected, cnode, c))
+  {
+      ret = connected_same_prefix (p, (c->address));
+      if(ret == 1)
+          return NULL;
+  }
   /* Allocate new connected address. */
   ifc = connected_new ();
   ifc->ifp = ifp;
