Index: hostapd-2.5/hostapd/Makefile
===================================================================
--- hostapd-2.5.orig/hostapd/Makefile
+++ hostapd-2.5/hostapd/Makefile
@@ -752,6 +752,7 @@ endif
 ifdef NEED_MD4
 ifdef CONFIG_INTERNAL_MD4
 OBJS += ../src/crypto/md4-internal.o
+CFLAGS += -DCONFIG_INTERNAL_MD4
 endif
 endif
 
Index: hostapd-2.5/src/crypto/crypto_openssl.c
===================================================================
--- hostapd-2.5.orig/src/crypto/crypto_openssl.c
+++ hostapd-2.5/src/crypto/crypto_openssl.c
@@ -93,6 +93,7 @@ static int openssl_digest_vector(const E
 }
 
 
+#ifndef CONFIG_INTERNAL_MD4
 #ifndef CONFIG_FIPS
 int md4_vector(size_t num_elem, const u8 *addr[], const size_t *len, u8 *mac)
 {
@@ -103,6 +104,7 @@ int md4_vector(size_t num_elem, const u8
 #endif
 }
 #endif /* CONFIG_FIPS */
+#endif /* CONFIG_INTERNAL_MD4 */
 
 
 void des_encrypt(const u8 *clear, const u8 *key, u8 *cypher)
