Index: net-snmp-5.8/include/net-snmp/library/snmp_openssl.h
===================================================================
--- net-snmp-5.8.orig/include/net-snmp/library/snmp_openssl.h
+++ net-snmp-5.8/include/net-snmp/library/snmp_openssl.h
@@ -70,11 +70,11 @@ extern          "C" {
 #endif
 #ifndef HAVE_X509_NAME_ENTRY_GET_OBJECT
     NETSNMP_IMPORT
-    ASN1_OBJECT *X509_NAME_ENTRY_get_object(const X509_NAME_ENTRY *ne);
+    ASN1_OBJECT *X509_NAME_ENTRY_get_object( X509_NAME_ENTRY *ne);
 #endif
 #ifndef HAVE_X509_NAME_ENTRY_GET_DATA
     NETSNMP_IMPORT
-    ASN1_STRING *X509_NAME_ENTRY_get_data(const X509_NAME_ENTRY *ne);
+    ASN1_STRING *X509_NAME_ENTRY_get_data( X509_NAME_ENTRY *ne);
 #endif
 #ifndef HAVE_X509_GET_SIGNATURE_NID
     NETSNMP_IMPORT
Index: net-snmp-5.8/snmplib/snmp_openssl.c
===================================================================
--- net-snmp-5.8.orig/snmplib/snmp_openssl.c
+++ net-snmp-5.8/snmplib/snmp_openssl.c
@@ -974,7 +974,7 @@ const unsigned char *ASN1_STRING_get0_da
 #endif
 
 #ifndef HAVE_X509_NAME_ENTRY_GET_OBJECT
-ASN1_OBJECT *X509_NAME_ENTRY_get_object(const X509_NAME_ENTRY *ne)
+ASN1_OBJECT *X509_NAME_ENTRY_get_object(X509_NAME_ENTRY *ne)
 {
     if (ne == NULL)
         return NULL;
@@ -983,7 +983,7 @@ ASN1_OBJECT *X509_NAME_ENTRY_get_object(
 #endif
 
 #ifndef HAVE_X509_NAME_ENTRY_GET_DATA
-ASN1_STRING *X509_NAME_ENTRY_get_data(const X509_NAME_ENTRY *ne)
+ASN1_STRING *X509_NAME_ENTRY_get_data(X509_NAME_ENTRY *ne)
 {
     if (ne == NULL)
         return NULL;
