Index: php-5.3.2/ext/phar/Makefile.frag
===================================================================
--- php-5.3.2.orig/ext/phar/Makefile.frag
+++ php-5.3.2/ext/phar/Makefile.frag
@@ -5,20 +5,10 @@ $(srcdir)/phar_path_check.c: $(srcdir)/p
 pharcmd: $(builddir)/phar.php $(builddir)/phar.phar
 
 PHP_PHARCMD_SETTINGS = -d 'open_basedir=' -d 'output_buffering=0' -d 'memory_limit=-1' -d phar.readonly=0 -d 'safe_mode=0'
+
 PHP_PHARCMD_EXECUTABLE = ` \
-	if test -x "$(top_builddir)/$(SAPI_CLI_PATH)"; then \
-		$(top_srcdir)/build/shtool echo -n -- "$(top_builddir)/$(SAPI_CLI_PATH) -n"; \
-		if test "x$(PHP_MODULES)" != "x"; then \
-		$(top_srcdir)/build/shtool echo -n -- " -d extension_dir=$(top_builddir)/modules"; \
-		for i in bz2 zlib phar; do \
-			if test -f "$(top_builddir)/modules/$$i.la"; then \
-				. $(top_builddir)/modules/$$i.la; $(top_srcdir)/build/shtool echo -n -- " -d extension=$$dlname"; \
-			fi; \
-		done; \
-		fi; \
-	else \
-		$(top_srcdir)/build/shtool echo -n -- "$(PHP_EXECUTABLE)"; \
-	fi;`
+		$(top_srcdir)/build/shtool echo -n -- "$(PHP_EXECUTABLE)";`
+
 PHP_PHARCMD_BANG = `if test -x "$(PHP_EXECUTABLE)"; then \
 		$(top_srcdir)/build/shtool echo -n -- "$(PHP_EXECUTABLE)"; \
 	else \
